### ISSUE TITLE
make build release doc workflow more robust

### ISSUE THREAD

**Issue by Remi-Gau** [nilearn/nilearn#4367]
see https://github.com/nilearn/nilearn/actions/workflows/release-docs.yml

seems a lot of the failures are related to failures to get the data

https://github.com/nilearn/nilearn/actions/runs/8613300789/job/23609732291

**Comment by bthirion** [comment]
This actually meas changing our policy toward the use of external data for examples.

**Comment by Remi-Gau** [comment]
well I think that this workflow does not try to cache any data between runs of the workflow like our usual doc build workflow does:

https://github.com/nilearn/nilearn/blob/cca69d661cf26977b693c605ac8ceec32fae7a31/.github/workflows/build-docs.yml#L139

**Comment by Remi-Gau** [comment]
Well the way it was done "before" (and still again now because we have not been able to make this workflow work for a release) would be to build the doc locally and push to update the website.

And I am pretty sure that when this was done, we did not delete the local nilearn data before starting the doc build, so we used "cached" data for our release build of the doc.

Don't see a reason why we don't want to do that in CI as well.
